### ISSUE TITLE
refactor!: Remove `src/options.ts`

### ISSUE THREAD

**Issue by danielbankhead** [googleapis/google-auth-library-nodejs#1658]
It's pretty legacy at this point and we can clean up this unnecessary code.

https://github.com/googleapis/google-auth-library-nodejs/blob/5f296c37390ba71aa3aa1e4259ff7c525eeaad7c/src/options.ts#L20-L32

**Comment by NewtonFernandis** [comment]
Hi @danielbankhead, I would like to work on this. Could you please assign this to me?

**Comment by danielbankhead** [comment]
Absolutely, take your time as we won't be able to merge this fix until the library's next major release (likely sometime next year).

**Comment by NewtonFernandis** [comment]
Cool 💯
